install.sh

```diff
@@ -370,80 +372,165 @@ if [ "$1" = "install" ]; then
-      else
-        _kernelname=$_basekernel.${_kernel_subver}-$_kernel_flavor
+    msg2 "Building kernel"
+    _timed_build make ${llvm_opt} -j ${_thread_num}
+    msg2 "Build successful"
+
+    if [ "$_STRIP" = "true" ]; then
+      echo "Stripping vmlinux..."
+      strip -v $STRIP_STATIC "vmlinux"
+    fi
+
+    _headers_folder_name="linux-$_kernelname"
+
+    msg2 "Removing unneeded architectures..."
+    for arch in arch/*/; do
+      [[ $arch = */x86/ ]] && continue
+      echo "Removing $(basename "$arch")"
+      rm -r "$arch"
+    done
+
+    msg2 "Removing broken symlinks..."
+    find -L . -type l -printf 'Removing %P\n' -delete
+
+    msg2 "Removing loose objects..."
+    find . -type f -name '*.o' -printf 'Removing %P\n' -delete
+
+    msg2 "Stripping build tools..."
+    while read -rd '' file; do
+      case "$(file -bi "$file")" in
+        application/x-sharedlib\;*)      # Libraries (.so)
+          strip -v $STRIP_SHARED "$file" ;;
+        application/x-archive\;*)        # Libraries (.a)
+          strip -v $STRIP_STATIC "$file" ;;
+        application/x-executable\;*)     # Binaries
+          strip -v $STRIP_BINARIES "$file" ;;
+        application/x-pie-executable\;*) # Relocatable binaries
+          strip -v $STRIP_SHARED "$file" ;;
+      esac
+    done < <(find . -type f -perm -u+x ! -name vmlinux -print0)
+
+    echo -e "\n\n"
+
+    msg2 "The installation process will run the following commands:"
+    echo "    # copy the patched and compiled sources to /usr/src/$_headers_folder_name"
+    echo "    sudo make modules_install"
+    echo "    sudo make install"
+    echo "    sudo dracut --force --hostonly ${_dracut_options} --kver $_kernelname"
+    echo "    sudo grub-mkconfig -o /boot/grub/grub.cfg"
+
+    msg2 "Note: Uninstalling requires manual intervention, use './install.sh uninstall-help' for more information."
+    read -p "Continue ? Y/[n]: " _continue
+
+    if ! [[ $_continue =~ ^(Y|y|Yes|yes)$ ]];then
+      exit 0
+    fi
+
+    msg2 "Copying files over to /usr/src/$_headers_folder_name"
+    if [ -d "/usr/src/$_headers_folder_name" ]; then
+      msg2 "Removing old folder in /usr/src/$_headers_folder_name"
+      sudo rm -rf "/usr/src/$_headers_folder_name"
+    fi
+    sudo cp -R . "/usr/src/$_headers_folder_name"
+    sudo rm -rf "/usr/src/$_headers_folder_name/.git"
+    cd "/usr/src/$_headers_folder_name"
+
+    msg2 "Installing modules"
+    if [ "$_STRIP" = "true" ]; then
+      sudo make modules_install INSTALL_MOD_STRIP="1"
+    else
+      sudo make modules_install
+    fi
+    msg2 "Removing modules from source folder in /usr/src/${_kernel_src_gentoo}"
+    sudo find . -type f -name '*.ko' -delete
+    sudo find . -type f -name '*.ko.cmd' -delete
+
+    msg2 "Installing kernel"
+    sudo make install
+    msg2 "Creating initramfs"
+    sudo dracut --force --hostonly ${_dracut_options} --kver $_kernelname
+    msg2 "Updating GRUB"
+    sudo grub-mkconfig -o /boot/grub/grub.cfg
```

Comment:
Some distros use `grub2-mkconfig` and similar naming conventions. You also might want to install to `/boot/efi/...` instead. Not sure why the script managed to run on Fedora though.

Reply:
Well, in any case, the `Generic` approach should not be used by rpm users. Hopefully the other approach of renaming the package names works so this is not needed.